File: svrScript.py
######import required libraries
import numpy as np # linear algebra
import pandas as pd # data processing
from sklearn.svm import SVR
from sklearn.model_selection import train_test_split #split data into train/test
from sklearn.preprocessing import StandardScaler
import matplotlib.pyplot as plt

df = pd.read_csv("data.csv") #load data values
#create train and test data
data = df.values
X = data[:, 1:]  # all rows, no label
y = data[:, 0]  # all rows, label only
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.25, random_state=42)

#Create SVR model 
model = SVR(C=1000, cache_size=500, epsilon=0.5, kernel='rbf')
#Fit the previously created model
print("Fitting SVR model this could take some time...")
model.fit(X_train, y_train)
#Add test values to the plot
trainlen = len(y_train)
plt.plot([i+trainlen for i in range(len(X_test[:,1]))], y_test)
#ADD predicted values for the test partition to the plot
plt.plot([i+trainlen for i in range(len(X_test[:,1]))], model.predict(X_test))
#show the plot
plt.show()


"""
####use the code below to perform different calculations for different values of epsiolon and C parameters
##try different epsilon values
epsilons = np.arange(1, 9)
scores = []
for e in epsilons:
	print("Checking epsilon: " + str(e))
	model.set_params(epsilon=e)
	model.fit(X_train, y_train)
	score=model.score(X_test, y_test)
	print("Score is: " +str(score))
	scores.append(score)
plt.plot(epsilons, scores)
plt.title("Epsilon effect")
plt.xlabel("epsilon")
plt.ylabel("score")
plt.show()

##try different Cost values
model.set_params(epsilon=5)
Cs = [1e0, 1e1, 1e2, 1e3]
scores = []
for c in Cs:
    model.set_params(C=c)
    model.fit(X_train, y_train)
    score=model.score(X_test, y_test)
    print("Score is: " +str(score))
    scores.append(score)
plt.plot(Cs, scores)
plt.title("C effect")
plt.xlabel("C")
plt.ylabel("score")
plt.show()
"""

File: svrScript.R
###Load required libraries
require(e1071)

###Load dataset
dataset <- read.csv("campus.csv")

##divide dataset into train and test
###set percentage for train set
perct <- 0.75
trainIndex <- round(length(dataset[,1])*perct)
train <- dataset [1:trainIndex,]
test <- dataset [-(1:trainIndex),]

##Create and fit the model with desired epsilon, kernel and Cost values
###in this example the value we want to predict is y and it is trained using all the rest of values
print("Fitting SVR model this could take some time...")
modelsvm <- svm(y ~ . ,data=train,cost=1000,kernel="radial",epsilon=0.5)
predictedsvm <- predict(modelsvm, newdata=test)

X11()  ####to show plots when running Rscript
plot(1:length(test[,1]),test$y,pch=16,ylab="Predicted value",xlab="")
points(1:length(test[,1]),predictedsvm,type="l",col="red")

#next two lines should be used when running Rscript to mantain the plot window opened
message("Press Return To Close window and continue")
invisible(readLines("stdin", n=1))


####In case we want to perform a search to find optimum values for epsilon and C
####we can perform a grid search

#tuneResult <- tune(svm, Y ~ .,  data = train,
#                   ranges = list(epsilon = seq(0,1,0.1), cost = 2^(2:9))
#)
#print(tuneResult)
## Draw the tuning graph
#plot(tuneResult)
